src/components/FixedGrid/Grid.js

```diff
@@ -23,24 +23,25 @@ const focusPosition = {
 function Grid(props) {
   const {
     className,
+    items,
+    order,
     style,
     setIsScroll,
     fixedRef,
     isBigScrollBtns,
     isNavigationButtonsOnTheSide,
-    gridState,
     ...other
   } = props;
-  const { items } = gridState;
 
-  const itemsPerPage = other.rows * other.columns;
-  const [pages, setPages] = useState(chunks(items, itemsPerPage));
+  const pages = useMemo(() => chunks(items, other.rows * other.columns), [
+    items,
+    other.rows,
+    other.columns
+  ]);
 
-  useEffect(
-    () => {
-      setPages(chunks(items, itemsPerPage));
-    },
-    [items, itemsPerPage]
+  const memoizedGridState = useMemo(
+    () => pages.map(pageItems => ({ items: pageItems, order })),
```

Comment:
Please, we should avoid copying the order of the first page to the others. It could cause confusion and bugs in the future